### PR TITLE
8315020: The macro definition for LoongArch64 zero build is not accurate.

### PR DESCRIPTION
**Pull request by theaoqi**
The `LOONGARCH` is not a macro that must be defined. It is not a C/C++ pre-defined macro and is also not pre-defined in libraries such as libffi. We'd better use a macro defined in jdk.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8315020](https://bugs.openjdk.org/browse/JDK-8315020): The macro definition for LoongArch64 zero build is not accurate. (**Bug** - P4)

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**) ⚠️ Review applies to [e944f3a5](https://git.openjdk.org/jdk/pull/15428/files/e944f3a5838b39063b5c00d1e457f0edbfd79868)
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15428/head:pull/15428` \
`$ git checkout pull/15428`

Update a local copy of the PR: \
`$ git checkout pull/15428` \
`$ git pull https://git.openjdk.org/jdk.git pull/15428/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15428`

View PR using the GUI difftool: \
`$ git pr show -t 15428`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15428.diff">https://git.openjdk.org/jdk/pull/15428.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15428#issuecomment-1693782212)

### CODE REVIEW

**Review by erikj79 — APPROVED** [review]

**Comment by theaoqi** [comment]
I am not sure if this need a second review. If yes, may I have it?

File: src/hotspot/os/linux/os_linux.cpp
Comment (RealFYang) [inline src/hotspot/os/linux/os_linux.cpp:1731]: Maybe we should also rename the `LOONGARCH` macro here at this line?https://github.com/openjdk/jdk/blob/master/src/hotspot/os/linux/os_linux.cpp#L1735

File: make/autoconf/platform.m4
Comment (RealFYang) [inline make/autoconf/platform.m4:585]: Would you mind moving the preceding two lines for RISCV32 to after PPC32? https://github.com/openjdk/jdk/blob/master/make/autoconf/platform.m4#L570C1-L571C34
We only have Zero support for RISCV32 for now. Thanks.

**Review by RealFYang — CHANGES_REQUESTED** [review]
Two minor nits.

**Comment by theaoqi** [comment]
Thanks for your advice, @RealFYang. I have updated the patch.

**Review by RealFYang — APPROVED** [review]

**Comment by theaoqi** [comment]
Thanks @erikj79 and @RealFYang for your review.
/integrate

**Comment by theaoqi** [comment]
Thank you, @RealFYang.

**Pull request closed without merging**
